src/main/java/com/patchworkmc/event/EventSubscriber.java

```diff
@@ -0,0 +1,59 @@
+package com.patchworkmc.event;
+
+import java.util.Objects;
+
+/**
+ * Object to hold information about a class that has at least one event subscriber.
+ */
+public class EventSubscriber {
```

Comment:
This class name is easy to confuse with `EventBusSubscriber` and `SubscribeEvent`. Maybe `SubscriberHolder` or `EventSubscriberClass`? There's probably a better name though.